### DISCUSSION TITLE
MSI MAG B550 Tomahawk Rebar does not work on the second videocard

### DISCUSSION THREAD

**Discussion by LsdMax** [terminatorul/NvStrapsReBar#71] [General]
MSI MAG B550 Tomahawk Rebar does not work on the second videocard.

![rebarb550_3](https://github.com/terminatorul/NvStrapsReBar/assets/44642519/c0c12ab7-a707-49bc-b34c-3e41d394526c)

Driver Version                            : 551.76
CUDA Version                              : 12.4
Attached GPUs                             : 2
GPU 00000000:04:00.0
    FB Memory Usage
        Total                             : 6144 MiB
        Reserved                          : 178 MiB
        Used                              : 0 MiB
        Free                              : 5966 MiB
    BAR1 Memory Usage
        Total                             : 256 MiB
        Used                              : 2 MiB
        Free                              : 254 MiB
GPU 00000000:2B:00.0
    FB Memory Usage
        Total                             : 8192 MiB
        Reserved                          : 205 MiB
        Used                              : 294 MiB
        Free                              : 7694 MiB
    BAR1 Memory Usage
        Total                             : 8192 MiB
        Used                              : 2 MiB
        Free                              : 8190 MiB

![rebarb550](https://github.com/terminatorul/NvStrapsReBar/assets/44642519/ac2b770e-d0ed-42f0-9996-1cbde769534f)

DXE Driver configuration:
isDirty:           0
OptionFlags:       0x02
nPciBarSize:       0
nGPUSelectorCount: 0
nGPUConfigCount:   2

GPUConfig1:    deviceID:        21C4
GPUConfig1:    subsysVendorID:  1458
GPUConfig1:    subsysDeviceID:  4012
GPUConfig1:    bus:             04
GPUConfig1:    device:          00
GPUConfig1:    function:        0
GPUConfig1:    BAR0 base:       0xF900'0000
GPUConfig1:    BAR0 top:        0xF9FF'FFFF

GPUConfig2:    deviceID:        1E81
GPUConfig2:    subsysVendorID:  10DE
GPUConfig2:    subsysDeviceID:  13A1
GPUConfig2:    bus:             2B
GPUConfig2:    device:          00
GPUConfig2:    function:        0
GPUConfig2:    BAR0 base:       0xFB00'0000
GPUConfig2:    BAR0 top:        0xFBFF'FFFF

nBridgeCount:      2

BridgeConfig1: vendorID:        1022
BridgeConfig1: deviceID:        43EA
BridgeConfig1: bus:             03
BridgeConfig1: device:          00
BridgeConfig1: function:        0
BridgeConfig1: secondary bus:   04

BridgeConfig2: vendorID:        1022
BridgeConfig2: deviceID:        1483
BridgeConfig2: bus:             00
BridgeConfig2: device:          03
BridgeConfig2: function:        1
BridgeConfig2: secondary bus:   2B

![resizeb550_2](https://github.com/terminatorul/NvStrapsReBar/assets/44642519/df52cb5f-cd6a-46fa-9a8d-e1128088136d)

Hello.
I used the latest version.
The driver does not want to work on the second video card.
Tell me what I need to change so that the extension starts working on the second video card.
I can build the driver ffc, and I have built it, and it is a copy of your file.
Thank you in advance.

**Comment by terminatorul** [comment]
I don't quite know what the issue can be at first sight. I have had multi-GPU setups that worked as expected.

Can you disable ReBAR in NvStrapsReBar.exe, reboot and then manually emable ReBAR only for the GTX 1660 (use input option 1, then P, then the 8 GB BAR size...) ?

If that works, then manually enable ReBAR for the second GPU (2080 Super)...

**Reply by terminatorul** [reply]
Failing that, you could even try the Override BAR Size Mask option ... unlikely to fix it, but you can try

**Comment by LsdMax** [comment]
Hello, I'm a programmer.
I see a clear problem that the driver in the BIOS cannot take into account the connection to the video card line.
I will write the truth. I was thinking of speeding up Nvidia CUDA on a second video card for nvidia physx.
The memory capacity is 8 gigabytes for PCI 3.0. Inclusion on only one card is not relevant. Enabling it gives 3-5 FPS in the video screensaver or when loading. Stream expansion only provides faster video processing. It's a pity that it doesn't work on the second video card.

**Reply by terminatorul** [reply]
So does it work if you enable ReBAR for the second GPU only ? Manually ?

The DXE driver status in your screenshot shows NvStrapsReBar is trying to enable ReBAR on the second GPU, only there is no response back... It is possible the BAR0 address saved in the configuration is somehow not available at the right time during boot, when NvStrapsReBar configures the new BAR size, but I have no way to be sure now. I will try to allocate the BAR0 address on-demand in a future version of NvStrapsReBar, but there is no way to know in advance if that will even work

**Comment by GregoriGT** [comment]
Добрый день. Извините что не в тему. Вопрос к LsdMax. У тебя в профиле указан сайт WotCheats.ru. Ты не в курсе, будет он работать? Спасибо.
